### PR TITLE
[Quant][Inductor] Enable quantization linear pattern fusion inside inductor

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106935
* __->__ #106934
* #106782
* #106781
* #105818

**Summary**
Enable lowering of quantized linear in Inductor

**Test plan**
python test/inductor/test_mkldnn_pattern_matcher.py -k test_qlinear_unary

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -3648,6 +3648,62 @@ def _original_deconv_weight_size(
...
+def _prepare_linear_fusion_create(
+    cls,
+    x: "TensorBox",
+    weight: "TensorBox",
+    bias: "TensorBox",
+):
+    """
+    This function is a helper function to prepare inputs, layout and constant args
+    for linear post-op fusion's create function. The function only supports the CPU device
+    since conv post-op fusion kernel is only supported on CPU right now.
```
Comment (jgong5) [inline torch/_inductor/ir.py:3660]:
```suggestion
    since linear post-op fusion kernel is only supported on CPU right now.
```

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -3648,6 +3648,62 @@ def _original_deconv_weight_size(
...
+                weight_fake,
+            )
+        else:
+            output = torch.ops.aten.mm.default(
+                x_fake,
+                weight_fake,
+            )
+        output_size = output.size()
+
+        req_stride_order = [0, 1]
```
Comment (leslie-fang-intel) [inline torch/_inductor/ir.py:3685]: Why we force un-contiguous input memory format here?

**Review by leslie-fang-intel — CHANGES_REQUESTED** [review]

File: torch/_inductor/ir.py (hunk 1 above)
Comment (Xia-Weiwen) [reply torch/_inductor/ir.py]: Thanks. It's fixed.

File: torch/_inductor/ir.py (hunk 2 above)
Comment (Xia-Weiwen) [reply torch/_inductor/ir.py]: Thanks. It's fixed.

**Review by leslie-fang-intel — APPROVED** [review]

File: torch/_inductor/ir.py [hunk 3]
```diff
@@ -4877,6 +4933,117 @@ def create(
         )
 
 
+class QLinearPointwisePT2E(ExternKernelAlloc):
```
Comment (eellison) [inline torch/_inductor/ir.py:4936]: if possible, would be great to make this logic more concise, like what you did with quantized_lowerings. If not could we move this and other quantized lowerings to `quantized_ir.py` or something along those lines ? does not have to be in this pr

**Review by eellison — APPROVED** [review]

File: torch/_inductor/ir.py (hunk 3 above)
Comment (Xia-Weiwen) [reply torch/_inductor/ir.py:4936]: Hi @eellison. Thanks for the suggestion. We think it difficult to make it more concise. We will move IR for quantized linear and conv to `quantized_ir.py` in a separate PR. How does that sound to you?

**Pull request closed without merging**